packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -177,10 +185,20 @@ export class SerializedStateManager {
 				snapshotSequenceNumber: attributes.sequenceNumber,
 			};
 
-			if (this.mc.config.getBoolean("Fluid.Container.enableOfflineSnapshotRefresh") === true)
-				this.refreshSnapshot ??= (async () => {
-					await this.refreshLatestSnapshot(supportGetSnapshotApi);
-				})();
+			// Fire and forget the refresh snapshot call (if enabled)
```

Comment:
Do we really want to fire-and-forget here?  Why not wait?  I'm concerned about the class variables being hard to predict in terms of their state since the call to `updateSnapshotAndProcessedOpsMaybe` could happen any time after this.

Reply:
(To be clear: It was fire-and-forget before, just less obvious IMO)

Reply:
So if it's not fire and forget, you'll slow down any sort of offline load. This is not part of the critical path. That would be my best guess.

Reply:
Yeah, makes perfect sense.  Probably getPendingLocalStateCore should await that promise though?  I'll follow up outside this PR